app/vue/contexts/CompetitionDetailsPageContext.js

```diff
@@ -246,6 +250,13 @@ export default class CompetitionDetailsPageContext extends BaseFuroContext {
           hooks: this.competitionLauncherHooks,
         })
 
+      // Top three and leaderboard invoke the same query.
+      // Make sure top three runs first to receive correct pagination result.
+      const currentPage = this.extractCurrentPage()
+      if (currentPage !== 1) {
+        await this.fetchTopThreeLeaderboardEntries()
+      }
```

Comment:
@hecticme (cc: @JYama321 @EG-easy )

The list item display and top three item display may have different display contents, so they cannot be used interchangeably.

```suggestion
      await this.fetchTopThreeLeaderboardEntries()
```